src/Jaeger.Core/Samplers/RemoteControlledSampler.cs

```diff
@@ -159,99 +114,101 @@ internal void UpdatePerOperationSampler(PerOperationSamplingStrategies samplingP
+            internal TimeSpan PollingInterval { get; private set; } = DefaultPollingInterval;
 
-            public Builder(string serviceName, ISamplingManager samplingManager)
+            public Builder(string serviceName)
             {
-                _serviceName = serviceName;
-                _samplingManager = samplingManager ?? throw new ArgumentNullException(nameof(samplingManager));
+                ServiceName = serviceName ?? throw new ArgumentNullException(nameof(serviceName));
+            }
+
+            public Builder WithSamplingManager(ISamplingManager samplingManager)
+            {
+                SamplingManager = samplingManager ?? throw new ArgumentNullException(nameof(samplingManager));
+                return this;
             }
 
             public Builder WithLoggerFactory(ILoggerFactory loggerFactory)
             {
-                _loggerFactory = loggerFactory;
+                LoggerFactory = loggerFactory ?? throw new ArgumentNullException(nameof(loggerFactory));
                 return this;
             }
 
             public Builder WithInitialSampler(ISampler initialSampler)
             {
-                _initialSampler = initialSampler;
+                InitialSampler = initialSampler ?? throw new ArgumentNullException(nameof(initialSampler));
                 return this;
             }
 
             public Builder WithMetrics(IMetrics metrics)
             {
-                _metrics = metrics;
+                Metrics = metrics ?? throw new ArgumentNullException(nameof(metrics));
                 return this;
             }
 
-            public Builder WithPollingInterval(int pollingIntervalMs)
+            public Builder WithPollingInterval(TimeSpan pollingIntervalMs)
             {
-                _pollingIntervalMs = pollingIntervalMs;
+                PollingInterval = pollingIntervalMs;
                 return this;
             }
 
             public RemoteControlledSampler Build()
             {
-                if (_loggerFactory == null)
+                if (LoggerFactory == null)
                 {
-                    _loggerFactory = NullLoggerFactory.Instance;
+                    LoggerFactory = NullLoggerFactory.Instance;
                 }
-                if (_initialSampler == null)
+                if (InitialSampler == null)
                 {
-                    _initialSampler = new ProbabilisticSampler();
+                    InitialSampler = new ProbabilisticSampler();
                 }
-                if (_metrics == null)
+                if (Metrics == null)
                 {
-                    _metrics = NoopMetricsFactory.Instance.CreateMetrics();
+                    Metrics = new MetricsImpl(NoopMetricsFactory.Instance);
                 }
-                return new RemoteControlledSampler(_serviceName, _samplingManager, _loggerFactory, _metrics, _initialSampler, _pollingIntervalMs);
+                return new RemoteControlledSampler(this);
```

Comment:
👍  I like that you just have a ctor for the `Builder`. Maybe we should use this everywhere where we only have a private ctor. Especially the `Tracer` ctor would become a lot easier and I'm sure some others as well.

Reply:
I agree - I'd prefer to do this in a separate PR though as this should also be done in Java.